lib/url.c

```diff
@@ -2403,13 +2403,18 @@ static CURLcode parse_proxy(struct Curl_easy *data,
   proxyinfo->proxytype = proxytype;
 
   /* Is there a username and password given in this proxy url? */
-  curl_url_get(uhp, CURLUPART_USER, &proxyuser, CURLU_URLDECODE);
-  curl_url_get(uhp, CURLUPART_PASSWORD, &proxypasswd, CURLU_URLDECODE);
+  uc = curl_url_get(uhp, CURLUPART_USER, &proxyuser, CURLU_URLDECODE);
+  if(uc && (uc != CURLUE_NO_USER))
```

Comment:
Why do two checks here? Wouldn't the latter one be sufficient as well?

Reply:
Because there might be a user in the URL and it might not. Both are fine, but all (other) errors are to be treated as errors.